azerothcore/azerothcore-wotlk
# fix(DB): Mux manascrambler's conditions

## Changes Proposed:
- Added conditions for both horde and alliance based on having the items
  on inventory and bank.
- Added his SmartAI to make the gossips function.

## Issues Addressed:

- Closes #8361

## SOURCE:

- Wowhead comments and classic sniffs
```
Misticka on 2008/04/24 (Patch 2.4.1)Subject: "Re: Mux Manascrambler"
Ok I found out on warcraft forums under quest section, you actually need to do the deliana quest first, 
and so this whole quest chain starts at her in IF. Mux will not talk to you until you finish her quest 
first and then sends you to mux. Then he talks to you and gives you quest.
```

## Tests Performed:

- Tested as Horde.

## How to Test the Changes:


1. .go c id 16014
2. Can't talk to the NPC
3. .q a 8923 (Horde) or 8922 (Alliance)
4. Talk to him -> No Gossips
5. .q a 8924
6. No Gossip for recovering the item
7. Delete item and check for gossip and recover the item
8. .q c 8924 and .q r 8924 (Removes the gossip option)
9. .q a 8977 (Alliance) 8978 (Horde)
10. Delete item and check for gossip to recover the item
11. Complete the quest and try again.

## Known Issues and TODO List:


None


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
